src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,53 @@
 package mate.academy;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newFixedThreadPool(5);
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        if (event == null) {
+            throw new IllegalArgumentException("Event cannot be null");
+        }
+        List<Callable<Void>> tasks = new ArrayList<>();
+        for (EventListener listener : listeners) {
+            tasks.add(() -> {
+                listener.onEvent(event);
+                return null;
+            });
+        }
+
+        try {
+            executorService.invokeAll(tasks);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
     }
 
     public void shutdown() {
-
+        executorService.shutdown();
+        try {
+            if (!executorService.awaitTermination(30, TimeUnit.SECONDS)) {
+                executorService.shutdownNow();
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+        listeners.clear();
```

Comment:
if you want you can also add some mechanism to not allow registering new listeners after we have initiated the shutdown, but that is up to you